### PR TITLE
feat(middleware-corr-ids): support for dynamodb streams

### LINKED ISSUE #67: Support DynamoDB streams as an event source
# This is a Feature Proposal

## Description

Similar to #39

### PR DESCRIPTION
**Pull request by gsingh1**
## What did you implement:

Closes #67 

## Todos:

- [x] Write tests
- [x] Write documentation
- [x] Fix linting errors
- [x] Make sure code coverage hasn't dropped
- [x] Provide verification config / projects / resources
- [x] Enable "Allow edits from maintainers" for this PR
- [x] Update the messages below

***Is this ready for review?:*** YES
***Is it a breaking change?:*** NO

### CODE REVIEW

**Comment by tume** [comment]
We really would like to see this feature merged. Anything I can help to push this forward? Testing maybe?

**Comment by theburningmonk** [comment]
@tume this feature is still actively being worked on, we're still figuring out the best way to cover all the use cases from the client's perspective (need to inject correlation IDs on PUT/BatchWrite/Transact/Update so a few edge cases to iron out)

File: packages/lambda-powertools-middleware-correlation-ids/package.json [hunk 1]
```diff
@@ -9,8 +9,9 @@
   "license": "MIT",
   "author": "Yan Cui",
   "dependencies": {
-    "@dazn/lambda-powertools-correlation-ids": "^1.8.2",
-    "@dazn/lambda-powertools-logger": "^1.9.0"
+    "@dazn/lambda-powertools-correlation-ids": "^1.5.0",
+    "@dazn/lambda-powertools-logger": "^1.8.0",
+    "aws-sdk": "^2.496.0"
```
Comment (gsingh1) [inline packages/lambda-powertools-middleware-correlation-ids/package.json]: Can make this a dev dep as it'll be available in the lambda runtime.

Comment (theburningmonk) [reply packages/lambda-powertools-middleware-correlation-ids/package.json]: Generally shouldn't depend on the version of AWS SDK in the runtime as it introduces subtle differences between what you tested and what's running in Lambda. The version in the Lambda runtime can also change without any action from you, so you lose immutable infra as well.

**Comment by theburningmonk** [comment]
@gsingh1 done the client bit, have a look. We should get the layer PR merged in first though, as it'd require further change to add the new dynamodb-client to the layer.

**Pull request merged**